services/DevHome.Services.WindowsPackageManager/Services/WinGetPackageInstaller.cs

```diff
@@ -51,29 +53,34 @@ public async Task<IWinGetInstallPackageResult> InstallPackageAsync(WinGetCatalog
-            var installResult = await InstallPackageInternalAsync(package, version, activityId);
-            var extendedErrorCode = installResult.ExtendedErrorCode?.HResult ?? HRESULT.S_OK;
-            var installErrorCode = installResult.GetValueOrDefault(res => res.InstallerErrorCode, HRESULT.S_OK); // WPM API V4
-
-            // 3. Report install result
-            _logger.LogInformation($"Install result: Status={installResult.Status}, InstallerErrorCode={installErrorCode}, ExtendedErrorCode={extendedErrorCode}, RebootRequired={installResult.RebootRequired}");
-            if (installResult.Status != InstallResultStatus.Ok)
+            return AsyncInfo.Run<IWinGetInstallPackageResult, WinGetInstallPackageProgress>(async (_, progress) =>
             {
-                throw new InstallPackageException(installResult.Status, extendedErrorCode, installErrorCode);
-            }
-
-            _logger.LogInformation($"Completed package installation for {packageId} from catalog {catalog.GetDescriptiveName()}");
-            TelemetryFactory.Get<ITelemetry>().Log("AppInstall_InstallSucceeded", Telemetry.LogLevel.Critical, new AppInstallResultEvent(package.Id, catalog.Catalog.Info.Id), activityId);
-            return new WinGetInstallPackageResult()
-            {
-                ExtendedErrorCode = extendedErrorCode,
-                RebootRequired = installResult.RebootRequired,
-            };
+                // 1. Find package
+                var package = await FindPackageOrThrowAsync(catalog, packageId);
+
+                // 2. Install package
+                _logger.LogInformation($"Starting package installation for {packageId} from catalog {catalog.GetDescriptiveName()}");
+                var install = InstallPackageInternalAsync(package, version, activityId);
+                install.Progress += (_, p) => progress.Report(new(p));
+                var installResult = await install;
+                var extendedErrorCode = installResult.ExtendedErrorCode?.HResult ?? HRESULT.S_OK;
+                var installErrorCode = installResult.GetValueOrDefault(res => res.InstallerErrorCode, HRESULT.S_OK); // WPM API V4
+
+                // 3. Report install result
+                _logger.LogInformation($"Install result: Status={installResult.Status}, InstallerErrorCode={installErrorCode}, ExtendedErrorCode={extendedErrorCode}, RebootRequired={installResult.RebootRequired}");
+                if (installResult.Status != InstallResultStatus.Ok)
+                {
+                    throw new InstallPackageException(installResult.Status, extendedErrorCode, installErrorCode);
+                }
+
+                _logger.LogInformation($"Completed package installation for {packageId} from catalog {catalog.GetDescriptiveName()}");
+                TelemetryFactory.Get<ITelemetry>().Log("AppInstall_InstallSucceeded", Telemetry.LogLevel.Critical, new AppInstallResultEvent(package.Id, catalog.Catalog.Info.Id), activityId);

```

Comment:
Instead of having different event names for an install failure and install success, have one event name and use either

1. different objects
2. The same object, but called with different constructors

to differentiate success/failure.

An example is to make the event name `AppInstall_InstallationStatus_Event` and have two objects

1. AppInstallEvent
2. AppInstallErrorEvent